R/agyw-integration.R

```diff
@@ -0,0 +1,1517 @@
+        "susceptible_sexnonreg" = sum(susceptible_sexnonreg * as.integer(age_group %in% age_groups)),
+        "susceptible_sexpaid12m" = sum(susceptible_sexpaid12m * as.integer(age_group %in% age_groups)),
+        "infections_nosex12m" = sum(infections_nosex12m * as.integer(age_group %in% age_groups)),
+        "infections_sexcohab" = sum(infections_sexcohab * as.integer(age_group %in% age_groups)),
+        "infections_sexnonreg" = sum(infections_sexnonreg * as.integer(age_group %in% age_groups)),
+        "infections_sexpaid12m" = sum(infections_sexpaid12m * as.integer(age_group %in% age_groups)),
+        "sexnonregplus" = sum(susceptible_sexnonreg, susceptible_sexpaid12m)/(population - plhiv),
+        .groups = "drop") %>%
+      dplyr::mutate(age_group = age_cat,
+                    nosex12m = susceptible_nosex12m/(population - plhiv),
+                    sexcohab = susceptible_sexcohab/(population - plhiv),
+                    sexnonreg = susceptible_sexnonreg/(population - plhiv),
+                    sexpaid12m = susceptible_sexpaid12m/(population - plhiv),
+                    incidence = (infections/(population - plhiv))*100,
+                    incidence_nosex12m = infections_nosex12m/susceptible_nosex12m,
+                    incidence_sexcohab = infections_sexcohab/susceptible_sexcohab,
+                    incidence_sexnonreg = infections_sexnonreg/susceptible_sexnonreg,
+                    incidence_sexpaid12m = infections_sexpaid12m/susceptible_sexpaid12m,
+                    prev_nosex12m = plhiv_nosex12m/(susceptible_nosex12m + plhiv_nosex12m),
+                    prev_sexcohab = plhiv_sexcohab/(susceptible_sexcohab + plhiv_sexcohab),
+                    prev_sexnonreg = plhiv_sexnonreg/(susceptible_sexnonreg + plhiv_sexnonreg),
+                    prev_sexpaid12m = plhiv_sexpaid12m/(susceptible_sexpaid12m + plhiv_sexpaid12m),
+                    rr_sexpaid12m = NA)
+  }
+
+  # Aggregate data
+  df2 <- dplyr::bind_rows(summarise_age_cat_female(df1, "Y015_024"),
+                          summarise_age_cat_female(df1, "Y025_049"),
+                          summarise_age_cat_female(df1, "Y015_049"))
+
+  # Calculate incidence
+  df3 <- dplyr::bind_rows(df1, df2) %>%
+    dplyr::mutate(incidence_cat = cut(incidence,
+                                      c(0, 0.3, 1, 3, 10^6),
+                                      labels = c("Low", "Moderate", "High", "Very High"),
+                                      include.lowest = TRUE, right = TRUE))
+
+  #' Check that sum of disaggregated infections is the same as total infections
+  sum_infections <- df3$infections_nosex12m + df3$infections_sexcohab + df3$infections_sexnonreg + df3$infections_sexpaid12m
+
+  # TO DO: Flag this to add in warning - stop please contact support (usually an issue with mapping boundaries)
+  # ADD IN WARNIING HERE
+  stopifnot(max(df3$infections - sum_infections) < 10^{-9})
```

Comment:
Do you want to do this as part of this PR? stopifnot's generally good to avoid as the error message they produce is not easy to understand by users

Reply:
Have converted into more informative warning.